service/worker/fx.go

```diff
@@ -89,6 +101,13 @@ func ConfigProvider(
 	)
 }
 
+func FrontendClientProvider(
```

Comment:
can we rename to `LocalFrontendClientProvider` and also add some comment that why `PublicClient` is actually `Local endpoint`?